### PR TITLE
[Backport] Update go.cmake to use PROJECT_SOURCE_DIR

### PR DESCRIPTION
**Pull request by justsmth**
### Description of changes: 
Backport of PR to fips-2022-11-02:
* #1484

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -729,6 +689,10 @@ if(NOT OPENSSL_NO_ASM AND CMAKE_OSX_ARCHITECTURES)
   list(GET CMAKE_OSX_ARCHITECTURES 0 CMAKE_SYSTEM_PROCESSOR)
 endif()
 
+if(MALLOC_FAILURE_TESTING)
+  add_definitions(-DBORINGSSL_MALLOC_FAILURE_TESTING)
+endif()
+
```
Comment (andrewhop) [inline CMakeLists.txt:695]: This looks unrelated to the original change.

**Review by darylmartin100 — APPROVED** [review]

**Review by andrewhop — APPROVED** [review]
